src/django_components/attributes.py

```diff
@@ -5,69 +5,42 @@
 from typing import Any, Dict, List, Mapping, Optional, Tuple
 
 from django.template import Context, Node
-from django.template.base import FilterExpression
 from django.utils.html import conditional_escape, format_html
 from django.utils.safestring import SafeString, mark_safe
 
-from django_components.template_parser import process_aggregate_kwargs
+from django_components.expression import Expression, safe_resolve
 
 HTML_ATTRS_DEFAULTS_KEY = "defaults"
 HTML_ATTRS_ATTRS_KEY = "attrs"
 
 
-def _default(val: Any, default_val: Any) -> Any:
-    return val if val is not None else default_val
-
-
 class HtmlAttrsNode(Node):
```

Comment:
Previously, when we passed aggregate kwargs (those `attrs:class="border-red"`) to `slot` or `html_attrs` tags, we processed these keys only at the render time.

I did an update to tag parsing, so that all our tags use the same common logic for pasring the inputs. And as part of that, also the aggregate kwargs are pre-processed up front. So a lot of logic from here could be removed.